no need to copy objects when annotated

William-Baker opened the pull request:
No need to copy annotations, this is expensive

William-Baker committed:
no need to copy objects when annotated

david-lindner commented on tracr/rasp/rasp.py:
```diff
@@ -157,10 +157,9 @@ def annotated(self: RASPExprT, **annotations) -> RASPExprT:
 
 def annotate(expr: RASPExprT, **annotations) -> RASPExprT:
   """Creates a new expr with added annotations."""
```
Should change this comment to reflect the new behavior (does not create a new expr anymore)

david-lindner reviewed (commented):
I suppose this does not improve the compilation performance but your program generation code? How large is the speedup?

I'm slightly hesitant about this change because it looks like something that might introduce silent bugs. I have to think a bit more whether something can break here.

If we make this change: we can also remove .copy and the dependency on copy.

William-Baker committed:
updated doc string and removed dependencies

William-Baker commented:
I've modified the doc string and removed the copy methods and dependency, at the time of removing this, it took around 30% of compilation time just on copy operations, however some were large programs

david-lindner reviewed (changes requested):
rasp_test currently fails -- pls fix before merging

William-Baker committed:
removed tests for copy() and copying when annotating

William-Baker commented:
I have removed the tests for copy() and copying when annotating, we could always keep the copy method and keep the copy tests?

david-lindner reviewed (changes requested):
Still makes rasp_to_craft_integration_test fail in a bunch of places. Can you look into this?

Also, can you please run all the tests in the future before making a PR? Thanks!
